output/typescript/types.ts

```diff
@@ -479,6 +479,33 @@ export interface InfoResponse {
   version: ElasticsearchVersionInfo
 }
 
+export interface KnnSearchRequest extends RequestBase {
+  index: Indices
+  routing?: Routing
+  body?: {
+    fields?: Fields
+    knn: KnnSearchKnnQuery
+  }
+}
+
+export interface KnnSearchResponse<TDocument = unknown> {
+  took: long
+  timed_out: boolean
+  _shards: ShardStatistics
+  hits: SearchHitsMetadata<TDocument>
+  fields?: Record<string, any>
+  max_score?: double
+}
+
+export interface KnnSearchKnnQuery {
```

Comment:
This naming is a little surprising since Knn appears twice (but not sure it will actually be presented to users?)